tools/ClangTidyAll.sh

```diff
@@ -42,14 +42,17 @@ EXIT_CODE=0
 
 # Check for run-clang-tidy, then run-clang-tidy-LLVM_VERSION, and
 # if we can't find those, then just loop over the files one at a time.
-if command -v run-clang-tidy > /dev/null 2>&1; then
-    run-clang-tidy -quiet ${THREADING_FLAG} -p ${BUILD_DIR} \
-                   ${ALL_SOURCE_FILES} || EXIT_CODE=1
-elif [ ! -z "${RUN_CLANG_TIDY_BIN}" ] && \
-         command -v ${RUN_CLANG_TIDY_BIN} > /dev/null 2>&1; then
-    ${RUN_CLANG_TIDY_BIN} -quiet ${THREADING_FLAG} -p ${BUILD_DIR} \
+RAN_CLANG_TIDY=1
```

Comment:
There's no reason for the values of this to be numbers, so how about 1 -> no and 0 -> yes to make the meaning clearer.  The test later will then have to change to `!=` (or `=`) to do string comparison.

Reply:
I have pushed a fixup that hopefully does what you imagined... Though not being well-versed in bash (comparisons especially still confuse me), please let me know if I got it wrong.

Reply:
You got it.  I'm fine with a squash.

(Comparisons are easy: you just learn the rules for perl and then do the opposite. And also remember that `=` is a synonym for `==` and both are in common use.  :P)

Reply:
Okay, thanks! I'll give @markscheel a chance to look at the fixups before I squash :+1: 

(haha, that's amazing :P )